src/tests/get_n_chars.pl

```diff
@@ -0,0 +1,207 @@
+:- module(get_n_chars_tests, []).
+:- use_module(test_framework).
+:- use_module(library(charsio)).
+:- use_module(library(iso_ext)).
+:- use_module(library(process)).
+:- use_module(library(format)).
+:- use_module(library(lists)).
+
+% Test 1: timeout=0 should behave exactly like get_n_chars/3
+test("timeout=0 equals get_n_chars/3", (
+    atom_chars('/bin/echo', Echo),
+    atom_chars('ABCDEFGHIJ', Content),
+    iso_ext:setup_call_cleanup(
+        process:process_create(Echo, [Content], [stdout(pipe(Out1))]),
+        iso_ext:setup_call_cleanup(
+            process:process_create(Echo, [Content], [stdout(pipe(Out2))]),
+            (
+                charsio:get_n_chars(Out1, 5, Chars1),
+                charsio:get_n_chars(Out2, 5, Chars2, 0),
+                Chars1 = Chars2
+            ),
+            close(Out2)
+        ),
+        close(Out1)
+    )
+)).
+
+% Test 2: Variable N with timeout=0
+test("variable N with timeout=0", (
+    atom_chars('/bin/echo', Echo),
+    atom_chars('Testing', Content),
+    iso_ext:setup_call_cleanup(
+        process:process_create(Echo, [Content], [stdout(pipe(Out1))]),
+        iso_ext:setup_call_cleanup(
+            process:process_create(Echo, [Content], [stdout(pipe(Out2))]),
+            (
+                charsio:get_n_chars(Out1, N1, Chars1),
+                charsio:get_n_chars(Out2, N2, Chars2, 0),
+                N1 = N2,
+                Chars1 = Chars2,
+                N1 = 8,
+                Chars1 = "Testing\n"
+            ),
+            close(Out2)
+        ),
+        close(Out1)
```

Comment:
This could be handled with a single `setup_call_cleanup/3`, pattern repeated many times